src/scat/parsers/qualcomm/qualcommparser.py

```diff
@@ -340,7 +341,10 @@ def run_diag(self, writer_qmdl = None):
                         writer_qmdl.write_cp(pkt + b'\x7e')
 
                     if parse_result is not None:
+                        if prev_parse_result and 'up' in prev_parse_result and 'up' in parse_result and prev_parse_result['up'] == parse_result['up']:
+                            break
                         self.postprocess_parse_result(parse_result)
+                        prev_parse_result = parse_result
```

Comment:
this is not really related to the topic of the PR, but seems to be necessary because I'm getting the same IP payload exactly twice (duplicated) when I run a trace on my device.